common/beerocks/btl/CMakeLists.txt

```diff
@@ -27,17 +23,19 @@ set(MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
 # Build the library
 add_library(${PROJECT_NAME} SHARED ${btl_sources})
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${${PROJECT}_VERSION_STRING} SOVERSION ${${PROJECT}_VERSION_MAJOR})
-target_link_libraries(${PROJECT_NAME} PUBLIC beerocks::bcl tlvf elpp beerocks::btlvf mapf::common)
-if (MapfTransport_FOUND)
-     target_link_libraries(${PROJECT_NAME} PRIVATE mapf::ieee1905_transport_messages)
+target_link_libraries(${PROJECT_NAME} PUBLIC bcl tlvf elpp btlvf common)
+if (NOT MSGLIB STREQUAL "None")
+     target_link_libraries(${PROJECT_NAME} PRIVATE ieee1905_transport_messages)
 endif()
 
 # Include paths
-target_include_directories(${PROJECT_NAME} 
+target_include_directories(${PROJECT_NAME}
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/>
     PRIVATE 
         ${MODULE_PATH} 
         ${MODULE_PATH}/../../../external
-        $<TARGET_PROPERTY:beerocks::bcl,INTERFACE_INCLUDE_DIRECTORIES>
+        $<TARGET_PROPERTY:bcl,INTERFACE_INCLUDE_DIRECTORIES>
```

Comment:
Why is this required anyway?
It makes sense for BWL and BPL as they are loaded dynamically, but for the BCL that it linked during build time, `target_link_libraries` should be enough, no?
If not, perhaps we should update the target to correctly set the INTERFACE_INCLUDE_DIRECTORIES.

Reply:
You're right - it's not needed, and neither does the external (which no longer exists nor the MODULE_PATH